test/graphql/create_group_mutation_test.rb

```diff
@@ -0,0 +1,274 @@
+    assert_equal 'New Group One Description', data['group']['description']
+
+    group = Group.last
+    # Top level group
+    assert_equal parent_group.to_global_id, group.parent.to_global_id
+    assert_equal 'new-group-one', group.path
+    assert_equal 'group-1/new-group-one', group.full_path
+  end
+
+  test 'createGroup mutation should work with valid params, parent group puid, and api scope token' do
+    parent_group = groups(:group_one)
+
+    result = IridaSchema.execute(CREATE_SUBGROUP_USING_GROUP_PUID_MUTATION,
+                                 context: { current_user: @user, token: @api_scope_token },
+                                 variables: { name: 'New Group One',
+                                              description: 'New Group One Description',
+                                              groupPuid: parent_group.puid })
+
+    assert_nil result['errors'], 'should work and have no errors.'
+
+    data = result['data']['createGroup']
+
+    assert_not_empty data, 'createGroup should be populated when no authorization errors'
+    assert_empty data['errors']
+    assert_not_empty data['group']
+
+    assert_equal 'New Group One', data['group']['name']
+    assert_equal 'New Group One Description', data['group']['description']
+
+    group = Group.last
+    # Top level group
+    assert_equal parent_group.puid, group.parent.puid
+    assert_equal 'new-group-one', group.path
+    assert_equal 'group-1/new-group-one', group.full_path
+  end
+
+  test 'createGroup mutation should work with valid params, no parent group id/puid, and api scope token' do
+    result = IridaSchema.execute(CREATE_GROUP_MUTATION,
+                                 context: { current_user: @user, token: @api_scope_token },
+                                 variables: { name: 'New Group One',
+                                              description: 'New Group One Description' })
+
+    assert_nil result['errors'], 'should work and have no errors.'
+
+    data = result['data']['createGroup']
+
+    assert_not_empty data, 'createGroup should be populated when no authorization errors'
+    assert_empty data['errors']
+    assert_not_empty data['group']
+
+    assert_equal 'New Group One', data['group']['name']
+    assert_equal 'New Group One Description', data['group']['description']
+
+    group = Group.last
+    # Top level group
+    assert_equal nil, group.parent
+    assert_equal 'new-group-one', group.path
+    assert_equal 'new-group-one', group.full_path
+  end
+
+  test 'createGroup mutation should work with custom path' do
+    result = IridaSchema.execute(CREATE_GROUP_MUTATION_WITH_CUSTOM_PATH,
+                                 context: { current_user: @user, token: @api_scope_token },
+                                 variables: { name: 'New Group Two',
+                                              description: 'New Group Two Description',
+                                              path: 'My Custom Path' })
```

Comment:
**nitpick (blocking):** With comment above this test should raise an error due to the path not passing validation.

Reply:
Added new tests for passing/failing with custom path in https://github.com/phac-nml/irida-next/pull/799/commits/e4202e61703f587b14214c1d94d38a032757fbc3